tests/integration/modules/test_vault.py

```diff
@@ -98,3 +99,121 @@ def test_delete_secret(self):
     def test_list_secrets(self):
         assert self.run_function('vault.write_secret', path='secret/my/secret', user='foo', password='bar') is True
         assert self.run_function('vault.list_secrets', arg=['secret/my/']) == {'keys': ['secret']}
+
+
+@destructiveTest
+@skipIf(not salt.utils.path.which('dockerd'), 'Docker not installed')
+@skipIf(not salt.utils.path.which('vault'), 'Vault not installed')
+class VaultTestCaseCurrent(ModuleCase):
+    '''
+    Test vault module against current vault
+    '''
+    count = 0
+
+    def setUp(self):
+        '''
+        SetUp vault container
+        '''
+        if self.count == 0:
+            config = '{"backend": {"file": {"path": "/vault/file"}}, "default_lease_ttl": "168h", "max_lease_ttl": "720h", "disable_mlock": true}'
+            self.run_state('docker_image.present', name='vault', tag='1.3.1')
+            self.run_state(
+                'docker_container.running',
+                name='vault',
+                image='vault:1.3.1',
+                port_bindings='8200:8200',
+                environment={
+                    'VAULT_DEV_ROOT_TOKEN_ID': 'testsecret',
+                    'VAULT_LOCAL_CONFIG': config,
+                }
+            )
+            time.sleep(5)
+            ret = self.run_function(
+                'cmd.retcode',
+                cmd='/usr/local/bin/vault login token=testsecret',
+                env={'VAULT_ADDR': 'http://127.0.0.1:8200'},
+            )
+            if ret != 0:
+                self.skipTest('unable to login to vault')
+            ret = self.run_function(
+                'cmd.retcode',
+                cmd='/usr/local/bin/vault policy write testpolicy {0}/vault.hcl'.format(FILES),
+                env={'VAULT_ADDR': 'http://127.0.0.1:8200'},
+            )
+            if ret != 0:
+                self.skipTest('unable to assign policy to vault')
+        self.count += 1
+
+    def tearDown(self):
+        '''
+        TearDown vault container
+        '''
+        def count_tests(funcobj):
+            return inspect.ismethod(funcobj) and funcobj.__name__.startswith('test_')
+        numtests = len(inspect.getmembers(VaultTestCaseCurrent, predicate=count_tests))
+        if self.count >= numtests:
+            self.run_state('docker_container.stopped', name='vault')
+            self.run_state('docker_container.absent', name='vault')
+            self.run_state('docker_image.absent', name='vault', force=True)
+
+    @flaky
```

Comment:
we are trying not to use this flaky decorator anymore. I know you will see it throughout our code, but we do have future plans to go through and clean up these tests so I don't want to add another flaky test to this list. These flaky tests also cause a lot of maintenance as they still do fail sometimes. Is this test indeed failing sometimes? If so is there something we could do to figure out why it is and make the test more dependable?

Reply:
These tests came with the flaky decorator by the time I looked at this. There **shouldn't** be anything particularly flaky about the operations tested. The integration test launches a pre-baked docker image, and runs some http calls against it. As long as the docker states are stable, and the hosting platform is able to run docker images okay, I think the flaky bit can be removed.

Reply:
Removed flaky'ness